core/src/main/java/com/sequenceiq/cloudbreak/service/sharedservice/DatalakeConfigProvider.java

```diff
@@ -74,17 +75,17 @@ public DatalakeResources collectAndStoreDatalakeResources(Stack datalakeStack, A
         try {
             return transactionService.required(() -> {
                 try {
-                    DatalakeResources datalakeResources = datalakeResourcesRepository.findByDatalakeStackId(datalakeStack.getId());
-                    if (datalakeResources == null) {
+                    Optional<DatalakeResources> datalakeResources = datalakeResourcesService.findByDatalakeStackId(datalakeStack.getId());
+                    if (datalakeResources.isPresent()) {
```

Comment:
Not sure but this might be wrong. It used to check for `null`. :)

Reply:
not much difference between checking an Optional for emptiness or an object for null, but making a return value as Optional, that significates that the given method **can** return something without value, which "forces" the caller to do something with this.

Reply:
Yeah, I am not at all against using `Optional` for that purpose. What I was referring to is that `isEmpty()` would be the right choice here if we want to act in case of the absence of DL resources. ❓

Reply:
Ohh, I see! There was a misunderstanding.. :) thank you, I'm going to fix it with the following commit